Danny's IS-10 Changes (merge into dannym-refactor) (#10)
AMWA-TV/nmos-authorization: docs/3.0. Discovery.md

Clients and Resource Servers MUST support discovering the Authorization Server through use of unicast DNS-SD service
discovery, as described in [RFC 6763][RFC-6763].

### Authorization Server Metadata Endpoint

The Server Metadata endpoint is described in [RFC 8414][RFC-8414]. The "/.well-known" URI suffix of "oauth-
authorization-server" MUST be used to identify the Authorization Server metadata endpoint. The value of the
Authorization Server metadata URL should take the form:

```

> { api_proto }://{ hostname }:{ port }/.well-known/oauth-authorization-server[/{ api_label }]

Reviewer: Could you indicate that the `api_` placeholders come from DNS-SD TXT records described further down?

comma-separated list of API versions supported by the server. For example: `v1.0,v1.1,v2.0`. There should be no
whitespace between commas, and versions should be listed in ascending order.

> This TXT record SHOULD NOT be used to form the URL of the authorization server endpoints, with the `api_label` record

Reviewer: ```suggestion
This TXT record MUST NOT be used to form the URL of the authorization server endpoints, with the `api_label` record
```

> used instead to locate the server metadata endpoint, and should instead be used as an indicator to the client of what

Reviewer: ```suggestion
used instead to locate the server metadata endpoint. The `api_ver` TXT record should instead be used solely as an indicator to the client of what
```

versions of this specification the authorization service supports.

#### pri

The DNS-SD advertisement MUST include a TXT record with key `pri` and an integer value. Servers MAY additionally
present a matching priority via the DNS-SD SRV record `priority` and `weight` as defined in [RFC 2782][RFC-2782]. The
TXT record should be used in favour of the SRV priority and weight where these values differ, in order to overcome
issues in the Bonjour and Avahi implementations. Values 0 to 99 correspond to an active NMOS Authorization Server API
(zero being the highest priority). Values 100+ are reserved for development work to avoid colliding with a live system.

#### api_label

The DNS_SD advertisement MAY include a TXT record of name `api_label`. The value of this TXT record is a string
relating to the "issuer identifier", as described in Section 3.1 of [RFC 8414][RFC-8414], and is appended to the server

> metadata URL. If this record is omitted or is equal to the empty string, the value of this record MUST be omitted from

Reviewer: ```suggestion
metadata URL. If this record is omitted or is equal to an empty string, the value of this record MUST be omitted from
```

> the path specifier for the well-known endpoint.

Reviewer: ```suggestion
the path specifier for the `.well-known` endpoint.
```

> This record value MAY be equal to the NMOS version supported by the advertised host.

Reviewer: ```suggestion
The `api_label` TXT record value MAY be equal to the NMOS API version supported by the advertised host.
```
